### PR TITLE
Whisper cli

### PR DESCRIPTION
**Pull request by niklasad1**
Hey,

This is an attempt to close [#6138](https://github.com/paritytech/parity/issues/6138)

EDIT:
I removed the tests for because it more or less requires duplication of the existing tests in ```rpc``` and ```whisper``` because these are private I can't just use them directly. 

It's alot of work re-implement (copy-paste) them so I don't think it is worth the effort. But maybe a couple of CLI arguments tests and some `rough` test that jsonrpc at least works!

Thoughts?

### CODE REVIEW

**Comment by 5chdn** [comment]
Nice. /cc @gballet FYI

Needs docs once merged.

File: whisper/cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+Whisper CLI.
+	Copyright 2017 Parity Technologies (UK) Ltd
+
+Usage:
+	whisper [options]
+	whisper [-h | --help]
+
+Options:
+	--whisper-pool-size SIZE       Specify Whisper pool size [default: 10].
+	-p, --port PORT                Specify which port to use [default: 8545].
```
Comment (rphmeier) [inline whisper/cli/src/main.rs]: maybe clarify that this is the RPC port

**Comment by rphmeier** [comment]
except for the small grumble, LGTM

**Comment by rphmeier** [comment]
I think we should use proper error handling; this failure is likely due to other bugs in the whisper, network, or RPC implementation and we should fix those rather than working around them.

**Comment by niklasad1** [comment]
@rphmeier 

I think it is good to go now, but I had to introduce an additional variable to `cancel/join` the `Decryption Worker Thread` because the `mspc` sends `Fn()` closure which I can't determine whether it is an actual message or to cancel the thread.

Alternately we could just skip joining threads and terminate!

**Comment by rphmeier** [comment]
> Alternately we could just skip joining threads and terminate!

Yep, also an option. The way it was before it would deadlock.

**Pull request merged**